src/js/angular/aclmanagement/model.js

```diff
@@ -4,94 +4,155 @@ export class ACListModel {
      * @param {ACRuleModel[]} aclRules
      */
     constructor(aclRules = []) {
-        this._aclRules = aclRules;
+        this._aclRules = new Map();
+        aclRules.forEach(rule => {
+            const scope = rule.scope;
+            if (!this._aclRules.has(scope)) {
+                this._aclRules.set(scope, []);
+            }
+            this._aclRules.get(scope).push(rule);
+        });
     }
 
-    size() {
-        return this.aclRules.length;
+
+    size(scope) {
+        return this._aclRules.get(scope).length;
     }
 
     /**
      * Creates a new ACRuleModel using provided values and appends it at the end of the list.
+     * @param {string} scope
+     * @param {string} policy
+     * @param {string} role
+     * @param {string} operation
      * @param {string} subject
      * @param {string} predicate
      * @param {string} object
      * @param {string} context
-     * @param {string} role
-     * @param {string} policy
+     * @param {string} plugin
      */
-    appendNewRule(subject, predicate, object, context, role, policy) {
-        this.aclRules.push(new ACRuleModel(subject, predicate, object, context, role, policy));
+    appendNewRule(scope, policy, role, operation, subject, predicate, object, context, plugin) {
+        const rule = ACRuleFactory.createRule(scope, policy, role, operation, subject, predicate, object, context, plugin);
+        if (!this._aclRules.has(scope)) {
+            this._aclRules.set(scope, []);
```

Comment:
Now this tends to propagate in multiple places which I don't like. I mean setting the default [] value and also getting the rules for given scope and defaulting to []. As far as the model is now a Map which is not typed, this leads to confusion and potential mistakes. That's why I tend to avoid using Map in plain js, but rather some custom models with methods.

Reply:
fixed the types of the Map and all is fine now